clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileSystem.java

```diff
@@ -473,7 +470,24 @@ private boolean deleteHelper(Path path) throws IOException {
     @Override
     public FileStatus[] listStatus(Path path) throws FileNotFoundException, IOException {
         LOG.debug("$$$$$$$$$$$$$$$$$$$$$$$$$$$$ List status is called for: {} $$$$$$$$$$$$$$$$$$$$$$$$$$$$", path.toString());
-        return new FileStatus[0];
+        ObjectLocation objectLoc = pathToObjectLocation(path);
+        try {
+            ObjectsApi objectsApi = lfsClient.getObjects();
+            ObjectStats objectStat = objectsApi.statObject(objectLoc.getRepository(), objectLoc.getRef(), objectLoc.getPath());
+            LakeFSFileStatus fileStatus = convertObjectStatsToFileStatus(objectLoc.getRepository(), objectLoc.getRef(), objectStat);
+            return new FileStatus[]{fileStatus};
+        } catch (ApiException e) {
+            if (e.getCode() != HttpStatus.SC_NOT_FOUND) {
+                throw new IOException("statObject", e);
+            }
+        }
+        List<FileStatus> fileStatuses = new ArrayList<>();
+        ListingIterator iterator = new ListingIterator(path, false, listAmount);
+        while (iterator.hasNext()) {
+            LocatedFileStatus fileStatus = iterator.next();
+            fileStatuses.add(fileStatus);
+        }
```

Comment:
Can you write some Java streams+collect magic to do this?

Reply:
tried a little - it is not pretty - this iterator implements RemoteIterator interface and not Iterator. Working with streams will require converting it to Iterator and make a stream out of it - if you can show me an easy way to do it I'll apply in the next PR.